File: scripts/create_issue.js
const axios = require('axios');
require('dotenv').config();

const createIssue = async () => {
  const url = `https://api.github.com/repos/${process.env.REPO_OWNER}/${process.env.REPO_NAME}/issues`;
  const token = process.env.GITHUB_TOKEN;

  const issueData = {
    title: 'Test Issue',
    body: 'This is a test issue created by the automated script.',
  };

  try {
    const response = await axios.post(url, issueData, {
      headers: {
        Authorization: `token ${token}`,
        'Content-Type': 'application/json',
      },
    });
    console.log('Issue created:', response.data.html_url);
  } catch (error) {
    console.error('Error creating issue:', error);
  }
};

createIssue();

File: scripts/create_issue.py
import os
import requests

def create_github_issue(repo, title, body=None, labels=None):
    token = os.getenv('GITHUB_TOKEN')
    if not token:
        raise ValueError('GITHUB_TOKEN is not set in environment variables')

    url = f'https://api.github.com/repos/{repo}/issues'
    headers = {'Authorization': f'token {token}'}
    data = {'title': title, 'body': body, 'labels': labels}

    response = requests.post(url, json=data, headers=headers)

    if response.status_code == 201:
        print('Successfully created issue')
    else:
        print(f'Failed to create issue: {response.content}')

    return response.json()